apps/dashboard/lib/trpc/routers/authorization/roles/permissions/schema-with-helpers.ts

```diff
@@ -0,0 +1,57 @@
+import { z } from "zod";
+
+export const LIMIT = 50;
+
+const RoleSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+});
+
+const PermissionSchema = z.object({
+  id: z.string(),
+  name: z.string(),
+  description: z.string().nullable(),
+  slug: z.string(),
+  roles: z.array(RoleSchema),
+});
+
+export const permissionsSearchPayload = z.object({
+  query: z.string().min(1, "Search query cannot be empty"),
+});
```

Comment:
nitpick: add the `trim` to this validation, then you can remove the redundant check:

```suggestion
export const permissionsSearchPayload = z.object({
  query: z.string().transform(val => val.trim()).pipe(
    z.string().min(1, "Search query cannot be empty")
  ),
});
```